src/main/java/org/eclipsefoundation/marketplace/service/impl/GuavaCachingService.java

```diff
@@ -47,43 +48,62 @@
 public class GuavaCachingService<T> implements CachingService<T> {
 	private static final Logger LOGGER = LoggerFactory.getLogger(GuavaCachingService.class);
 
-	@ConfigProperty(name = "cache.max", defaultValue = "2500")
+	@ConfigProperty(name = "cache.max", defaultValue = "10000")
 	long maxSize;
-	@ConfigProperty(name = "cache.ttl.access", defaultValue = "21600")
+	@ConfigProperty(name = "cache.ttl.access", defaultValue = "900")
 	long ttlAccess;
-	@ConfigProperty(name = "cache.ttl.write", defaultValue = "86400")
+	@ConfigProperty(name = "cache.ttl.write", defaultValue = "900")
 	long ttlWrite;
 
 	// actual cache object
 	Cache<String, T> cache = null;
+	Map<String, Long> ttl;
 
 	@PostConstruct
 	public void init() {
+		this.ttl = new HashMap<>();
+		// create cache with configured settings that maintains a TTL map
 		cache = CacheBuilder
-				.newBuilder()
-				.maximumSize(maxSize)
-				.expireAfterAccess(ttlAccess, TimeUnit.SECONDS)
-				.expireAfterWrite(ttlWrite, TimeUnit.SECONDS)
-				.build();
+					.newBuilder()
+					.maximumSize(maxSize)
+					.expireAfterAccess(ttlAccess, TimeUnit.SECONDS)
+					.expireAfterWrite(ttlWrite, TimeUnit.SECONDS)
+					.removalListener(not -> ttl.remove(not.getKey()))
+					.build();
+
 	}
 
 	@Override
 	public Optional<T> get(String id, RequestWrapper params, Callable<? extends T> callable) {
 		Objects.requireNonNull(id);
 		Objects.requireNonNull(params);
 		Objects.requireNonNull(callable);
-		
-		String cacheKey = getCacheKey(id, params);		
+
+		String cacheKey = getCacheKey(id, params);
 		try {
+			// check if the cache is bypassed for the request
+			if (params.isCacheBypass()) {
+				return Optional.of(callable.call());
```

Comment:
When the request bypass the cache, don't you want to update the cache anyway with the freshest value?

Reply:
The idea for this method is that certain calls shouldn't ever be cached, like the random results. We could always update if present, that way if a value has been cached before it would update it, but not for things like random results. If we start caching the random results, it will just create churn in the mem cache, which likely would hurt performance

Reply:
I get that, but I dislike the fact that the client control the caching behavior, could be a open breach for DoS attack. If some call should not update the cache, make it a server side thing only,

Reply:
The bypass attribute is set as an attribute, rather than a parameter. In this way, it should only be controlled by the server rather than the client. The only way in which the client "controls" this is if they set up one of the conditions that set the internal request attribute. If I'm misunderstanding the attributes in the request, just let me know and I'll flip it around!

Reply:
Thanks for the explanation. I've read the code too quickly and though it was a query parameter.